helper/resource/testing.go

```diff
@@ -533,13 +548,35 @@ func Test(t testing.T, c TestCase) {
 // testProviderConfig takes the list of Providers in a TestCase and returns a
 // config with only empty provider blocks. This is useful for Import, where no
 // config is provided, but the providers must be defined.
-func testProviderConfig(c TestCase) string {
+func testProviderConfig(c TestCase) (string, error) {
 	var lines []string
+	var requiredProviders []string
 	for p := range c.Providers {
 		lines = append(lines, fmt.Sprintf("provider %q {}\n", p))
 	}
+	for p, v := range c.ExternalProviders {
+		if _, ok := c.Providers[p]; ok {
```

Comment:
I think you need to check factories as well here?

Reply:
Factories don't produce a provider block :man_shrugging: